src/somark.rs

```diff
@@ -0,0 +1,35 @@
+//! somark - abstraction to be used only on linux
+//!
+//! on other platforms it's noop without memory footprint
+
+use socket2::SockRef;
+
+#[derive(Copy, Clone, Debug)]
+pub struct SoMark {
```

Comment:
You may want to add a `#[repr(transparent)]` to enure there is no overhead.
https://doc.rust-lang.org/nomicon/other-reprs.html#reprtransparent

Reply:
good point. changed.